test/e2e/alertmanager_test.go

```diff
@@ -28,86 +28,89 @@ import (
 )
 
 func TestAlertmanagerCreateDeleteCluster(t *testing.T) {
+	t.Parallel()
+
+	ctx := testFramework.NewTestCtx(t)
+	defer ctx.CleanUp(t)
+	ctx.BasicSetup(t, framework.KubeClient)
```

Comment:
This just creates a namespace. In favor of explicitness, I'd just rename this to `ctx.CreateNamespace` and actually have it return the namespace name.

Below we are calling functions expecting a namespace name with `ctx.Id`, which seems somewhat cryptic unless you know that `Id` is always equal to a namespace created during the basic setup.

```go
ns := ctx.CreateNamespace(test) // creates namespace `<ctx.Id>-test`
// ...
framework.DeleteAlertmanagerAndWaitUntilGone(ns, name)
```

This seems a bit clearer overall. What I find important is that it translates seamlessly for tests that need to create multiple namespaces, which would not have an obvious solution from looking at tests that just need `BasicSetup`.

```go
ns1 := ctx.CreateNamespace("test1")
ns2 := ctx.CreateNamespace("test2")
// .. do whatever needed with either namespace
```